### ISSUE TITLE
w3sper: Reorganize and Clarify StakeInfo Structure

### ISSUE THREAD

**Issue by nortonandreev** [dusk-network/rusk#3124]
## Description  
The current structure and naming of properties within the `StakeInfo` object could use improvement for clarity and alignment with intuitive expectations. Specifically:

### Confusion with `amount` property  
- The `amount` property of `StakeInfo` gets initialized only after an active stake is made.  
- This behavior is not immediately clear from the name. The term `amount` suggests a simpler numeric property, not an object type or a property tied to the stake's state.

### Counterintuitive `value` vs. `locked` distinction  
- Within the `amount` object, "value" represents locked funds, while "locked" represents soft-slashed tokens.  
- This distinction is not intuitive and can lead to confusion, as "locked" is expected to describe locked funds rather than slashed tokens.

### Misleading `eligibility` property name  
- The `eligibility` property, which defines the block at which the stake matures, does not clearly convey its purpose.  
- A more descriptive name might better represent its function.

## Suggestions  
- **Rename `amount` property**  
   Consider renaming `amount` to something like `stakeDetails` or `stakeState` to better reflect its purpose as an object type with detailed information.

- **Clarify `value` and `locked` usage**  
   Revisit the naming of "value" and "locked" within the `amount` object to ensure they intuitively represent their meaning. For example:  
   - Rename "value" to `lockedValue` or `lockedFunds`.  
   - Rename "locked" to `softSlashed`.

- **Improve `eligibility` naming**  
   Change `eligibility` to something more descriptive, such as `maturityBlock` or `unlockBlock`.

- **Consider breaking down `StakeInfo` further**  
   - Separate the maturity information (`eligibility`) from the financial details (`amount`) to enhance modularity and clarity.  

## Expected Outcome  
A more intuitive `StakeInfo` structure with descriptive property names that align with their usage, improving code readability and reducing potential confusion among developers.

**Comment by ZER0** [comment]
I would probably close this issue as WONTFIX:

- The structure here represents the structure of the contract as well. It would be too misleading to have completely different naming for all these properties, especially since some of them are also used in consensus.
- Once the library is typed/documented, this won't be much of an issue, and `stake.amount.value` is totally legitimate naming.
- `locked` is not only related to slashing; it represents the amount currently "locked." For example, a stake's top-up after the eligibility period will incur a penalty of 10% of the top-up value, which will be "locked" and therefore appear there. The `value` property does not represent a "locked" value but the value of the amount currently staked. That is not locked; otherwise, you wouldn't be able to unstake it.
- The `amount` cannot be renamed to `StakeDetails` or something similar because `StakeInfo` already serves that purpose. The structure already details the stake itself, while `StakeAmount` refers to the details of the amount.
- As mentioned, most terms are also used in the stake contract and consensus and/or in the blockchain world (e.g., `eligibility`), so I wouldn't try to change them either.
- We also shouldn't break down the structure further, and not only to maintain parity with the contract / consensus, but because it wouldn't make sense. For example, `eligibility` and `locked` make sense only if there is a `value`, which is why they're grouped together, while the counters for faults are not, and that's why they're outside of `StakeAmount`.
